docs/proposals/block-storage-time-series-deletion.md

```diff
@@ -0,0 +1,247 @@
+
+One important thing to note regarding this rewrite tool is that it should not be used at the same time as when another compactor is touching a block. If the tool is run at the same time as compaction on a particular block, it can cause overlap and the data marked for deletion can already be part of the compacted block. To mitigate such issues, these are some of the proposed solutions:
+
+Option 1: Only apply the deletion once the blocks are in the final state of compaction.
+
+Pros:
+- Simpler implementation as everything is contained within the DeletedSeriesCleaner.
+
+Cons:
+- Might have to wait for a longer period of time for the compaction to be finished. 
+    - This would mean the earliest time to be able to run the deletion would be once the last time from the block_ranges in the [compactor_config](https://cortexmetrics.io/docs/blocks-storage/compactor/#compactor-configuration) has passed. By default this value is 24 hours, so only once 24 hours have passed and the new compacted blocks have been created, then the rewrite can be safely run. 
+
+
+
+
+Option 2: For blocks that still need to be compacted further after the deletion request cancel period is over, the deletion logic can be applied before the blocks are compacted. This will generate a new block which can then be used instead for compaction with other blocks. 
+
+Pros:
+- The deletion can be applied earlier than the previous options. 
+    - Only applies if the deletion request cancel period is less than the last time interval for compaction is. 
+Cons:
+- Added coupling between the compaction and the DeletedSeriesCleaner. 
+- Might block compaction for a short time while doing the deletion. 
+
+
+
+Once all the applicable blocks have been rewritten without the deleted data, the deletion request state moves to `Syncing`. Once the syncing time period is over, the state will advance to `Processed` state and the tombstone will no longer be used.  
+
+
+
+#### Handling failed/unfinished delete jobs:
+
+Deletions will be completed and the tombstones will be deleted only when the DeletedSeriesCleaner iterates over all blocks that match the time interval and confirms that they have been re-written without the deleted data.  Otherwise, it will keep creating the markers indicating which blocks are remaining for deletion. In case of any failure that causes the deletion to stop, any unfinished deletions will be resumed once the service is restarted. The series deletion markers will remain in the bucket until the new blocks are created without the deleted data. Meaning that the compactor will continue to process the blocks for deletion that are remaining according to the deletion markers. 
```

Comment:
By "deletion markers", do you mean "tombstones" / "deletion jobs"?

> The series deletion markers will remain in the bucket until the new blocks are created without the deleted data.

How do we detect this condition?

Reply:
My bad, the series deletion markers were proposed as part of a previous iteration of the proposal. After some feedback, I've decided they are not necessary. I thought I removed them from the proposal but missed this section. I've updated this section to reflect that. Thanks.